steward/src/cellars.rs

```diff
@@ -23,12 +23,17 @@ pub async fn get_gas_price() -> Result<U256, Error> {
     CellarGas::etherscan_standard().await.map_err(|e| e.into())
 }
 
-fn parse_cellar_id(cellar_id: &str) -> CellarId {
-    let parts = cellar_id.split_at(cellar_id.chars().position(|c| c == ':').unwrap());
+fn parse_cellar_id(cellar_id: &str) -> Result<CellarId, String> {
+    let delimiter = match cellar_id.chars().position(|c| c == ':') {
```

Comment:
I don't think you need to go through all of this, this should suffice:

https://doc.rust-lang.org/std/primitive.str.html#method.split

Reply:
Improved and added unit tests